pandas/tests/frame/test_operators.py

```diff
@@ -451,6 +451,18 @@ def test_arith_flex_frame(self):
         with tm.assert_raises_regex(NotImplementedError, 'fill_value'):
             self.frame.add(self.frame.iloc[0], axis='index', fill_value=3)
 
+    def test_arith_flex_zero_len_raises(self):
+        # GH#19522 passing fill_value to frame flex arith methods should
+        # raise even in the zero-length special cases
+        ser = pd.Series([])
+        df = pd.DataFrame([[1, 2], [3, 4]])
+
+        with tm.assert_raises_regex(NotImplementedError, 'fill_value'):
+            df.add(ser, fill_value='E')
+
+        with tm.assert_raises_regex(NotImplementedError, 'fill_value'):
+            ser.to_frame().sub(df[0], axis=None, fill_value=3)
```

Comment:
just create the required frame here rather than using ``ser.to_frame()``

Reply:
Sure.